### PR TITLE
CSUB-1345: Migrate away from Azure in favor of Linode

### PR DESCRIPTION
**Pull request by atodorov**
NOTE: opening against release-polkadot-v1.1.0-patch (instead of release-polkadot-v1.1.**1**-patch) b/c this is what gluwa/creditcoin3 & gluwa/creditcoin3-next are compiled against).

### CODE REVIEW

**Comment by atodorov** [comment]
FTR I think that the failure reported by `gluwa / cargo-test` is rustc going out of memory. I can also reproduce this locally.

**Comment by atodorov** [comment]
FTR at https://github.com/gluwa/polkadot-sdk/actions/runs/11741349059/job/32710035675?pr=5 I see 2 unit tests failing, which however are passing over at https://github.com/gluwa/polkadot-sdk/actions/runs/11738728011/job/32701907195?pr=3. 

From what I can tell starting branch/commit for both PR#3 and PR#5 is the same and there are no changes related to the polkadot-sdk code base itself. IDK what contributes to the different outcomes here, nor I'm sure what do we want to do with that information.

**Comment by beqaabu** [comment]
> FTR at https://github.com/gluwa/polkadot-sdk/actions/runs/11741349059/job/32710035675?pr=5 I see 2 unit tests failing, which however are passing over at https://github.com/gluwa/polkadot-sdk/actions/runs/11738728011/job/32701907195?pr=3.
> 
> From what I can tell starting branch/commit for both PR#3 and PR#5 is the same and there are no changes related to the polkadot-sdk code base itself. IDK what contributes to the different outcomes here, nor I'm sure what do we want to do with that information.

The difference between the two runs is the `--release` flag. 
https://github.com/gluwa/polkadot-sdk/actions/runs/11741349059/workflow?pr=5#L143 - fails
https://github.com/gluwa/polkadot-sdk/actions/runs/11738728011/workflow?pr=3#L163 - passes

I have been able to reproduce the same thing locally. compiling without `--release` flag makes the tests green

File: .github/workflows/gluwa.yml [hunk 1]
```diff
@@ -153,46 +137,85 @@ jobs:
       - uses: Swatinem/rust-cache@v2
 
       - name: Run tests
-        uses: actions-rs/cargo@v1
+        uses: gluwa/cargo@dev
         with:
           command: test
-          args: --features runtime-benchmarks
+          args: --release --features runtime-benchmarks
```
Comment (beqaabu) [inline .github/workflows/gluwa.yml]: this change is breaking

**Review by beqaabu — CHANGES_REQUESTED** [review]

File: .github/workflows/gluwa.yml (hunk 1 above)
Comment (atodorov) [reply .github/workflows/gluwa.yml]: Also confirmed in https://github.com/gluwa/polkadot-sdk/actions/runs/11820013472/job/32931491897?pr=4 when I cherry-pick the same commit into PR#4.

**Review by beqaabu — APPROVED** [review]

**Pull request merged**